e2e/single-cluster/oci_registry_test.go

```diff
@@ -654,4 +669,142 @@ var _ = Describe("Single Cluster Deployments using OCI registry", Label("oci-reg
+					Expect(err).ToNot(HaveOccurred())
+					for _, s := range secrets.Items {
+						Expect(s.Name).ToNot(HavePrefix("s-"))
+					}
+				})
+			})
+		})
+	})
+
+	When("applying a gitrepo and then applying a new one with different contents and the same name", func() {
+		Context("containing a valid helm chart and later some raw contents", func() {
+			BeforeEach(func() {
+				experimentalValue = true
+				insecureSkipTLS = true
+				deployDefaultSecret = true
+				deploySpecificSecretName = ""
+				forceDefaultReference = ""
+				forceUserDefinedReference = ""
+				useReaderAsWriter = false
+			})
+
+			It("redeploys the gitrepo and only the last OCI artifact and related secret should remain", func() {
+				var bundle fleet.Bundle
+				By("creating the bundle", func() {
+					k8sclient.GetObjectShouldSucceed(clientUpstream, "sample-simple-chart-oci", env.Namespace, &bundle)
+				})
+				By("setting the ContentsID field in the bundle", func() {
+					Expect(bundle.Spec.ContentsID).To(ContainSubstring("s-"))
+					contentsID = bundle.Spec.ContentsID
+				})
+				By("waiting until the helm chart is deployed", func() {
+					Eventually(func(g Gomega) {
+						var cm corev1.ConfigMap
+						k8sclient.GetObjectShouldSucceed(clientUpstream, "test-simple-chart-config", "default", &cm)
+					}).Should(Succeed())
+				})
+				By("creating an OCI artifact with the expected name", func() {
+					secretKey := client.ObjectKey{Name: contentsID, Namespace: env.Namespace}
+					opts, err := ocistorage.ReadOptsFromSecret(context.TODO(), clientUpstream, secretKey)
+					Expect(err).ToNot(HaveOccurred())
+					_, err = ocistorage.NewOCIWrapper().PullManifest(context.TODO(), opts, contentsID)
+					Expect(err).ToNot(HaveOccurred())
+				})
+				By("deploying a gitrepo with the same name and different content", func() {
+					gitrepo := path.Join(tempDir, "gitrepo.yaml")
+					err := testenv.Template(gitrepo, testenv.AssetPath(asset), struct {
+						OCIRegistrySecret string
+						Branch            string
+					}{
+						deploySpecificSecretName,
+						"oci-storage-git-repo-changed",
+					})
+					Expect(err).ToNot(HaveOccurred())
+
+					out, err := k.Apply("-f", gitrepo)
+					Expect(err).ToNot(HaveOccurred(), out)
+				})
+				var bundle2 fleet.Bundle
+				previousContentsID := contentsID
+				By("updating the bundle, setting ContentsID to a different value", func() {
+					Eventually(func(g Gomega) {
+						k8sclient.GetObjectShouldSucceed(clientUpstream, "sample-simple-chart-oci", env.Namespace, &bundle2)
+						g.Expect(bundle2.Spec.ContentsID).To(ContainSubstring("s-"))
+						g.Expect(previousContentsID).ToNot(Equal((bundle2.Spec.ContentsID)))
+						contentsID = bundle2.Spec.ContentsID
+					}).Should(Succeed())
+				})
+				By("checking that the previous oci bundle key was deleted", func() {
+					var secret corev1.Secret
+					k8sclient.ObjectShouldNotExist(clientUpstream, previousContentsID, env.Namespace, &secret, false)
```

Comment:
nit: since `secret` does not need to be used, this could probably be shortened as:
```suggestion
					k8sclient.ObjectShouldNotExist(clientUpstream, previousContentsID, env.Namespace, &corev1.Secret{}, false)
```